src/drupal7.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Drupal 7 phpass variant using SHA-512 and hashes cut at 258 bits.
+ *
+ * This software is Copyright (c) 2012 magnum, and it is hereby released to the
+ * general public under the following terms:  Redistribution and use in source
+ * and binary forms, with or without modification, are permitted.
+ *
+ * These are 8 byte salted hashes with a loop count that defines the number
+ * of loops to compute. Drupal uses 258 bits of the hash, this is a multiple of
+ * 6 but not 8. I presume this is for getting unpadded base64. Anyway we store
+ * an extra byte but for now we will only compare 256 bits. I doubt that will
+ * pose any problems. Actually I'm not quite sure the last bits end up correct
+ * from the current version of get_binary().
+ *
+ * Moved common stuff from drupal7_fmt_plug.c into drupal7.h
+ * - Denis Burykin, 2018
```

Comment:
I think this file should be called `drupal7_common.h` to be consistent with other similar files we have.